bayesian/api_v1.py

```diff
@@ -416,6 +417,56 @@ def post():
         return {'status': 'success'}
 
 
+class UserIntent(ResourceWithSchema):
+    method_decorators = [login_required]
+
+    @staticmethod
+    def get(user, ecosystem):
+        if user is None:
+            raise HTTPError(400, error="Expected user name in the request")
+
+        if ecosystem is None:
+            raise HTTPError(400, error="Expected ecosystem in the request")
+
+        s3 = StoragePool.get_connected_storage('S3ManualTagging')
+        # get user data
+        try:
+            result = s3.fetch_user_data(user, ecosystem)
+        except Exception as exc:
```

Comment:
Please catch only `botocore.exceptions.ClientError` exception. This way we will return 404 every time that is not something we would like to do - this error can be confusing. Leave 500 in cases when something unpredictable happened (e.g. S3 is down).

```python
import botocore

...

except botocore.exceptions.ClientError:
    current_app.logger.exception("Failed to fetch data for the user {u}".format(u=user))
    raise HTTPError(404, error="Failed to fetch data for the user {u}, ecosystem {e}".format(u=user, e=ecosystem))
```

Reply:
I've updated error hanlding.

Reply:
this makes sense

Reply:
Please respect review comments.